genai-perf/genai_perf/inputs/retrievers/synthetic_data_retriever.py

```diff
@@ -105,44 +112,49 @@ def _generate_stateless_entries(self, use_prefix_prompts: bool) -> List[DataRow]
         data_rows = []
 
         for _ in range(self.config.num_dataset_entries):
-            row = self._create_data_row(use_prefix_prompts)
-            row.images = [
-                self._generate_image() for _ in range(self.config.batch_size_image)
-            ]
+            row = self._create_data_row()
+            row.texts = self._generate_prompts(use_prefix_prompts)
+            row.images = self._generate_images()
             data_rows.append(row)
 
         return data_rows
 
-    def _create_data_row(
-        self, use_prefix_prompts: bool, session_id: str = ""
-    ) -> DataRow:
-        row = DataRow(texts=[], images=[], optional_data={}, payload_metadata={})
+    def _create_data_row(self, session_id: str = "") -> DataRow:
```

Comment:
why did you take the work out of create_data_row()?

Reply:
Because the function name and it's actual behavior has a mismatch. It says create the data row but it's actually creating data row _AND_ generating prompts.